lib/miq_pglogical.rb

```diff
@@ -7,22 +7,41 @@ class MiqPglogical
   PUBLICATION_NAME = 'miq'.freeze
   ALWAYS_EXCLUDED_TABLES = %w(ar_internal_metadata schema_migrations repl_events repl_monitor repl_nodes).freeze
 
-  def self.with_connection_error_handling
-    retry_attempted ||= false
-    yield
-  rescue PG::ConnectionBad
-    raise if retry_attempted
-
-    pglogical(true)
-    retry_attempted = true
-    retry
-  end
+  module ConnectionHandling
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def with_connection_error_handling
+        retry_attempted ||= false
+        yield
+      rescue PG::ConnectionBad
+        raise if retry_attempted
+
+        pglogical(true)
+        retry_attempted = true
+        retry
+      end
+
+      def pglogical(refresh = false)
+        @pglogical = nil if refresh
+        @pglogical ||= PG::LogicalReplication::Client.new(pg_connection)
+      end
+    end
 
-  def initialize
-    @pg_connection = ApplicationRecord.connection.raw_connection
+    def pglogical(refresh = false)
+      self.class.pglogical(refresh)
+    end
   end
 
-  delegate :subscriber?, :to => :pglogical
+  include ConnectionHandling
+
+  # :nodec:
+  #
+  # Note:  Don't use a delegate with this method, since we want the
+  # .with_connection_error_handling to wrap this methods
+  def subscriber?
+    self.class.with_connection_error_handling { pglogical.subscriber? }
```

Comment:
Necessary change that fixes some specs that were failing:

https://travis-ci.com/github/ManageIQ/manageiq-cross_repo-tests/jobs/452658848#L2687-L2719